VPCameraActor _ VPCameraActor 추가

sanghun219 opened the pull request:
# 세부사항
1.  Actor에 카메라 생성 및 조작
  - Actor에대해 고정, 상대적인 Transform을 가지도록 생성, FieldofView나 카메라의 위치 회전을 설정가능.
2. Actor의 외부적 요인
  - Actor당 붙을 수 있는 최대 카메라 수, 2개 이상의 카메라에 대한 카메라 전환 기능, 기즈모나 UI적 요소에 의한
    Transform 및 카메라 설정에 대한 것들은 외부 Class에서 조작.

bangbangdog219 committed:
VPCameraActor _ VPCameraActor 추가

# 세부사항
1.  Actor에 카메라 생성 및 조작
  - Actor에대해 고정, 상대적인 Transform을 가지도록 생성, FieldofView나 카메라의 위치 회전을 설정가능.
2. Actor의 외부적 요인
  - Actor당 붙을 수 있는 최대 카메라 수, 2개 이상의 카메라에 대한 카메라 전환 기능, 기즈모나 UI적 요소에 의한
    Transform 및 카메라 설정에 대한 것들은 외부 Class에서 조작.

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+#include <GameFramework/PlayerController.h>
+#include <Kismet/KismetMathLibrary.h>
+#include <GameFramework/Character.h>
+#include <EngineUtils.h>
```
상위 모듈별로 묶어서 인클루드 정리해줄 것 :
ex)
#include <Camera/PlayerCameraManager.h>
#include <Camera/CameraComponent.h>
#include <Camera/CameraActor.h>
이런 식으로

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+		this->ProjectionMode = ProjectionMode;
+		MyCameraComponent->ProjectionMode = (ECameraProjectionMode::Type)ProjectionMode;
+		this->SetActorTransform(Transform);
+		this->MyCameraComponent->bConstrainAspectRatio = false;
```
굳이 this 넣을 필요없으니 정리해주셈.

Minusi replied:
아래 코드에 들어간 this도 전부

Minusi commented on UEPrototype/Source/UEPrototype/Public/ActorInfo/VPCameraActor.h:
```diff
@@ -0,0 +1,137 @@
+class USpringArmComponent;
+class USceneComponent;
+
+UENUM(BlueprintType)
```
계속 얘기하지만 각 형식(Type)에는 무엇을 하기 위해 생성되어 있는 형식인지에 대한 설명을 적어주셈.
enum class, class, property, function 모두

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+		return;
+
+	//타겟 카메라로 전환됨.
+	PlayerController->SetViewTargetWithBlend(this, ViewTarget.BlendTime, ViewTarget.BlendFunc, ViewTarget.BlendExp
```
PC에 대한 유효성 검사가 누락됨.

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+
+		PlayerCameraMgr = GetPlayerCameraMgr();
+		MyCameraComponent = GetCameraComponent();
+
```
PC, CameraComp에 대한 유효성 검사가 누락됨. 사용하기 전에는 반드시 유효한지 검사할 것.

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+// Gizmo를 붙여줘야함 나중에 Transform대신에 GizmoTransform이 달려야함
+void AVPCameraActor::AdjustCameraSetting_Perspective(FTransform Transform, FPerspective Perspective)
+{
+	MyCameraComponent->SetRelativeTransform(Transform);
```
CameraComp 유효성 검사 누락

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+// Gizmo를 붙여줘야함 나중에 Transform대신에 GizmoTransform이 달려야함
+void AVPCameraActor::AdjustCameraSetting_OrthoGraphic(FTransform Transform, FOrthoGraphic OrthoGraphic)
+{
+
```
CameraComp 유효성 검사 누락

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+
+void AVPCameraActor::ReturnPlayerCamera()
+{
+	//플레이어 카메라로 즉시 전환
```
PC 유효성 검사 누락

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+
+void AVPCameraActor::ZoomIn()
+{
+	if (MyCameraComponent->FieldOfView > 0)
```
CameraComp 유효성 검사 누락

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+
+void AVPCameraActor::ZoomOut()
+{
+	if (MyCameraComponent->FieldOfView < 90)
```
CameraComp 유효성 검사 누락

Minusi commented on UEPrototype/Source/UEPrototype/Private/ActorInfo/VPCameraActor.cpp:
```diff
@@ -0,0 +1,130 @@
+		UE_LOG(LogTemp, Log, TEXT("GetCameraActor :: GEngine is nullptr"));
+		return nullptr;
+	}
+
```
PC 유효성 검사 누락

Minusi commented on UEPrototype/Source/UEPrototype/Public/ActorInfo/VPCameraActor.h:
```diff
@@ -0,0 +1,137 @@
+
+	// 카메라 컴포넌트 캐싱을 위한 레퍼런스
+	UPROPERTY(BlueprintReadOnly, Category = "VPCameraActor", meta = (AllowPrivateAccess = "true"))
+	UCameraComponent* MyCameraComponent;
```
변수명이 무엇을 의미하는 지 짐작할 수 없음. 변수명을 읽는 것만으로 이 프로퍼티가 무슨 역할인지 알 수 있도록 적을 수 있도록 하셈. 캐시가 목적이면 "CameraComponentCached" 또는 "CachedCameraComp(onent)"가 대안이 될 수 있음.

Minusi commented on UEPrototype/Source/UEPrototype/Public/ActorInfo/VPCameraActor.h:
```diff
@@ -0,0 +1,137 @@
+	APlayerController* PlayerController;
+
+	// Perspective , Orthographic
+	EPROJECTION_MODE ProjectionMode;
```
이 변수를 포함한 4개의 변수에 대해서 언리얼 리플렉션 시스템에 노출시키지 않는 이유가 있음? 있으면 얘기해주고 없으면 UPROPERTY() 기술하는 게 맞음.

bangbangdog219 committed:
VPCameraActor _ VPCameraActor 추가

#세부사항
 1. Actor에 카메라 생성 및 조작
   - Actor에대해 고정, 상대적인 Transform을 가지도록 생성, FieldofView나 카메라의 위치 회전을 설정가능.
  2. Actor의 외부적 요인
    - Actor당 붙을 수 있는 최대 카메라 수, 2개 이상의 카메라에 대한 카메라 전환 기능, 기즈모나 UI적 요소에 의한
Transform 및 카메라 설정에 대한 것들은 외부 Class에서 조작.

Minusi committed:
Merge branch 'master' into Interaction
